### PR TITLE
A new front-end interface that uses configuration files

### PR DESCRIPTION
**Pull request by sqazi**
The configuration files are represented as python dictionaries.  A couple of different payload executors are supported.  One generically converts rows (generated by a row generator) into payloads.  Another does some analysis on row hammer tolerance.

### CODE REVIEW

**Review by jedrzejboczar — CHANGES_REQUESTED** [review]
It looks cool. I was able to run it on my board without problems.
I left some comments with suggestions regarding the code.

I was also thinking that maybe we could modularize the directory structure.
One suggestion would be to just group payload generators and row generators into separate directories (as I guess we want to add more later), for example (`->` denotes new names).
```
payload_generators/
    payload_generator.py  -> __init__.py
    hammer_tolerance_payload_generator.py -> hammer_tolerance.py
    row_list_payload_generator.py -> row_list.py
row_generators/
    row_generator.py -> __init__.py
    even_row_generator.py -> even_row.py
example_hammer_tolerance.cfg
example_row_list_payload.cfg
row_mapping.py
run_config.md
run_config.py
```

We could go one step further, but I am not sure about it:
```
config_runner/
    __init__.py  # Empty file
    payload_generators/
        payload_generator.py  -> __init__.py
        hammer_tolerance_payload_generator.py -> hammer_tolerance.py
        row_list_payload_generator.py -> row_list.py
    row_generators/
        row_generator.py -> __init__.py
        even_row_generator.py -> even_row.py
    example_hammer_tolerance.cfg
    example_row_list_payload.cfg
    row_mapping.py
    run_config.md
    run_config.py -> run.py
```
In this one the name `config_runner` is just an initial thought.

And as a minor suggestion `run_config.md` could be named in uppercase as that's a great summary and user will most likely want to read it first. We should also definitely link to it from the main readme.

Feel free to only follow the suggestions that you think are sensible and I think we can merge it.

Also, if you think the formatting from https://github.com/antmicro/litex-rowhammer-tester/pull/39 would be ok, then we can merge your PR first, then re-run the formatting in https://github.com/antmicro/litex-rowhammer-tester/pull/39 and merge it.

**Comment by kgugala** [comment]
@sqazi can you rebase this PR to resolve conflicts and also run code formatting check on it?

File: rowhammer_tester/scripts/run_config.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    return dict(row_errors)
+
+def main():
+    valid_keys = set(["payload_generator", "payload_generator_config",
+                      "inversion_divisor", "inversion_mask", "row_pattern"])
+    parser = argparse.ArgumentParser()
+    parser.add_argument("config_file", type=open)
+    args = parser.parse_args()
+    config = eval(args.config_file.read())
```
[the remark this answers is not in the document]
Comment (sqazi) [reply rowhammer_tester/scripts/run_config.py:50]: One thought that crossed my mind is that there is a small side benefit of using python: you could do things like specifying range() in the future.  But, it's not a big deal.  The JSON looks pretty similar to how the current config looks, and it doesn't sound like a lot of work to make the necessary adjustments, so I don't mind changing over.

**Comment by sqazi** [comment]
@kgugala and @jedrzejboczar, thanks for the quick review.  Scanning over, I like most of the suggestions and will take them, and I will rebase this PR, resolve conflicts and run the formatter.

**Pull request closed without merging**